src/GAP/iso_oscar_gap.jl

```diff
@@ -70,9 +70,12 @@ function _iso_oscar_gap_field_finite_functions(FO::Union{Nemo.fpField, Nemo.FpFi
    return (f, finv)
 end
 
+Nemo.lift(R::ZZRing, x::ZZRingElem) = x
+Nemo.lift(R::ZZRing, x::Integer) = ZZRingElem(x)
+
 function _iso_oscar_gap_field_finite_functions(FO::Union{FqPolyRepField, FqField, fqPolyRepField}, FG::GAP.GapObj)
    p = characteristic(FO)
-   d = degree(FO)
+   d = degree(FO)::Int
```

Comment:
Somehow it wasn't able to infer the return value of `degree` here. I am looking into a proper fix now